cmd/baton-confluence/config.go

```diff
@@ -4,36 +4,42 @@ import (
 	"context"
 	"fmt"
 
-	"github.com/conductorone/baton-sdk/pkg/cli"
-	"github.com/spf13/cobra"
+	"github.com/conductorone/baton-sdk/pkg/field"
+	"github.com/spf13/viper"
 )
 
-// config defines the external configuration required for the connector to run.
-type config struct {
-	cli.BaseConfig `mapstructure:",squash"` // Puts the base config options in the same place as the connector options
+var (
+	apiKeyField = field.StringField(
+		"api-key",
+		field.WithDescription("The api key for your Confluence account. ($BATON_API_KEY)"),
+	)
+	domainUrl = field.StringField(
+		"domain-url",
+		field.WithDescription("The domain url for your Confluence account. ($BATON_DOMAIN_URL)"),
+	)
+	usernameField = field.StringField(
+		"username",
+		field.WithDescription("The username for your Confluence account. ($BATON_USERNAME)"),
+	)
+)
 
-	ApiKey    string `mapstructure:"api-key"`
-	DomainUrl string `mapstructure:"domain-url"`
-	Username  string `mapstructure:"username"`
+// configurationFields defines the external configuration required for the connector to run.
+var configurationFields = []field.SchemaField{
+	apiKeyField,
+	domainUrl,
+	usernameField,
```

Comment:
I think you can get rid of validateConfig() by using WithRequired(true) in the field declarations.